src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -177,36 +179,67 @@ final class TrialService
-            // user already enrolled; don't re-enroll
-            case Some(TrialStates.Enrolled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are already enrolled in a free trial."))
-            // user enabled (eligible) for trial, enroll!
-            case Some(TrialStates.Enabled) => {
-              // build the new state that we want to persist to indicate the user is enrolled
-              val now = Instant.now
-              val expirationDate = now.plus(FireCloudConfig.Trial.durationDays, ChronoUnit.DAYS)
-              val enrolledStatus = status.copy(
-                state = Some(TrialStates.Enrolled),
-                enrolledDate = now,
-                expirationDate = expirationDate
-              )
-              thurloeDao.saveTrialStatus(userInfo.id, userInfo, enrolledStatus) map { _ =>
-                // TODO: add user to free-trial billing project / create said project if necessary
-                RequestComplete(NoContent)
-              }
+    thurloeDao.getTrialStatus(userInfo.id, userInfo) flatMap {
+      // can't determine the user's trial status; don't enroll
+      case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 10)"))
+      case Some(status) =>
+        status.state match {
+          // user already enrolled; don't re-enroll
+          case Some(TrialStates.Enrolled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are already enrolled in a free trial. (Error 20)"))
+          // user enabled (eligible) for trial, enroll!
+          case Some(TrialStates.Enabled) => enrollUserInternal(userInfo, status)
+          // user in some other state; don't enroll
+          case Some(TrialStates.Disabled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 30)"))
+          case Some(TrialStates.Terminated) => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 40)"))
+          case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 50)"))
+        }
+    }
+  }
+
+  private def enrollUserInternal(userInfo: UserInfo, status: UserTrialStatus): Future[PerRequestMessage] = {
+
+    val projectId = "fccredits-hafnium-peach-3794" // TODO: Make this status.billingProjectName when GAWB-2911 lands
+    val saToken: WithAccessToken = AccessToken(OAuth2BearerToken(googleDAO.getTrialBillingManagerAccessToken))
+
+    // 1. Check that the assigned Billing Project exists and contains exactly one member, the SA we used to create it
+    rawlsDAO.getProjectMembers(projectId)(saToken) flatMap { members: Seq[RawlsBillingProjectMember] =>
+      if (members.map(_.email.value) != Seq(googleDAO.getTrialBillingManagerEmail)) {
+        Future(RequestCompleteWithErrorReport(InternalServerError, "We could not process your enrollment. (Error 60 - billing project exists but cannot be used)"))
```

Comment:
For this story: log the project name and the project's members, so we can analyze the error

as a TODO for a follow-on story: claim a new project for this user and try adding to that new project.